Dockerfile

```diff
@@ -501,6 +501,96 @@ RUN \
 
 # =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
 
+FROM sdk-libc as sdk-go-fips
+
+ARG ARCH
+ARG TARGET="${ARCH}-bottlerocket-linux-gnu"
+ARG SYSROOT="/${TARGET}/sys-root"
+ARG GOVER="1.21.4"
+ARG AWS_LC_FIPS_VER="2.0.2"
+
+USER root
+RUN DEBIAN_FRONTEND=noninteractive dnf install -y golang cmake xz wget ca-certificates clang python curl git
+
+USER builder
+WORKDIR /home/builder/sdk-go-fips
+COPY ./hashes/go /home/builder/hashes
+RUN \
+  sdk-fetch /home/builder/hashes && \
+  tar --strip-components=1 -xf go${GOVER}.src.tar.gz && \
+      rm go${GOVER}.src.tar.gz
+
+WORKDIR /home/builder/sdk-go-fips
+COPY patches/go-fips/* ./
+RUN git init && git apply --whitespace=nowarn *.patch
+
+# Build aws-lc crypto modules
+USER root
+WORKDIR /boring/boringssl
```

Comment:
We should not run builds as root:
```suggestion
# Build aws-lc crypto modules
WORKDIR /home/builder/boring
```

Reply:
The go boringcrypto build script makes these assumptions about the paths where the source is: https://github.com/golang/go/blob/27316739ac249bb07d47990ceb4013d442fd28dd/src/crypto/internal/boring/build-goboring.sh#L28-L31

Since the script is intended to run its own docker build environment. I'll go ahead and try patching the script.